--- rpy2-3.2.5.orig/rpy2/tests/rinterface/test_embedded_r.py	2019-12-18 18:36:55.000000000 -0300
+++ rpy2-3.2.5/rpy2/tests/rinterface/test_embedded_r.py	2020-01-27 15:59:05.555284099 -0300
@@ -71,13 +71,14 @@
         rinterface.embedded.set_initoptions(('aa', '--verbose',
                                              '--no-save'))
 
+def _init_r():
+    from rpy2 import rinterface
+    rinterface.initr()
+
 
 def test_initr():
-    def init_r():
-        from rpy2 import rinterface
-        rinterface.initr()
     preserve_hash = True
-    proc = multiprocessing.Process(target=init_r,
+    proc = multiprocessing.Process(target=_init_r,
                                    args=(preserve_hash,))
     proc.start()
     proc.join()
