From 87af00804fb23d9b7c006540d7d552809f864b5b Mon Sep 17 00:00:00 2001
From: Flavio Martins <flavio.f.martins@tecnico.ulisboa.pt>
Date: Thu, 10 Jul 2025 17:24:27 +0100
Subject: [PATCH] Focus on windows build.

---
 setup.py | 21 ++++++++++++++++++---
 1 file changed, 18 insertions(+), 3 deletions(-)

diff --git a/setup.py b/setup.py
index 979c46e46..0d4bee289 100644
--- a/setup.py
+++ b/setup.py
@@ -8,6 +8,7 @@
     sys.exit(1)
 
 import enum
+import importlib
 import os
 import shutil
 import subprocess
@@ -21,9 +22,23 @@
 from setuptools._distutils.errors import CCompilerError, DistutilsExecError, DistutilsPlatformError
 
 import setuptools.command.build
+import setuptools.command.build_ext
 import setuptools.command.install
 
-import importlib
+link_args = ['-static-libgcc',
+             '-static-libstdc++',
+             '-Wl,-Bstatic,--whole-archive',
+             '-lwinpthread',
+             '-Wl,--no-whole-archive']
+
+
+class build_ext(setuptools.command.build_ext.build_ext):
+    def build_extensions(self):
+        if self.compiler.compiler_type == 'mingw32':
+            for e in self.extensions:
+                e.extra_link_args = link_args
+        super().build_extensions()
+
 
 # spec = importlib.util.spec_from_file_location('rpy2', './rpy2/__init__.py')
 # rpy2 = importlib.util.module_from_spec(spec)
@@ -206,8 +221,8 @@ def run(self):
 setup(
     cffi_modules=cffi_modules,
     ext_modules=ext_modules,
-    cmdclass=dict(build=build),
-    long_description=long_description, 
+    cmdclass={'build': build, 'build_ext': build_ext},
+    long_description=long_description,
     # List of pacakges moved to project.toml.
     # TODO: package_data should be be moved to project.toml when setuptools
     # supports it (see note in project.toml).
