--- rpy2-2.9.1.orig/rpy/robjects/pandas2ri.py	2017-11-04 14:54:54.000000000 -0200
+++ rpy2-2.9.1/rpy/robjects/pandas2ri.py	2018-03-05 15:50:10.706865983 -0300
@@ -144,7 +144,7 @@
 def get_timezone():
     """ Return the system's timezone settings. """
     etc_timezone = '/etc/timezone'
-    timezone = None
+    timezone = pytz.timezone('UTC')
     if os.path.exists(etc_timezone):
         try:
             with open(etc_timezone) as fh:
@@ -153,10 +153,8 @@
                 timezone = pytz.timezone(tz)
             except pytz.UnknownTimeZoneError:
                 warnings.warn('Unknown time zone in %s, using UTC.' % etc_timezone)
-                timezone = 'UTC'
         except IOError:
             warnings.warn('Unable to read date in %s, using UTC.' % etc_timezone)
-            timezone = 'UTC'
     else:
         warnings.warn('No file %s' % etc_timezone)
     return timezone
